### PR TITLE
Re-structuring + Webpack config + Minimal Popup UI

### LINKED ISSUE #5: [Chrome Extension] Implement popup UI
Implement popup UI
- HTML
- CSS

### PR DESCRIPTION
**Pull request by junhur**
1. Re-structure:
  - extension popup entry point renamed to index.tsx
  - popup.tsx is a standalone component
  - html & css separated
2. Webpack config:
  - Dynamically get all `.ts` & `.tsx` files and set them as webpack entry 
3. Minimal Popup UI:
  - Top app bar with yubaba & placeholder settings icon
  - Title of the article, followed by summary
  - Loading Spinner added but doesn't show at the moment because of short response time (@sungho-cho for visibility)
     - Might have to adjust the position of the loading spinner. 

* Could possibly do another re-structure if we feel that we need extra react components
  (dedicated components directory?)

### CODE REVIEW

**Comment by chorongi** [comment]
Great job on the UI component!

File: extension/webpack.config.js [hunk 1]
```diff
@@ -1,46 +1,58 @@
...
+    }
+});
+
 module.exports = {
-    entry: {
-        popup: './src/ts/popup.tsx',
-        background: './src/ts/background.ts',
-        contentScript: './src/ts/contentScript.ts',
-    },
+    entry,
```
Comment (sungho-cho) [inline extension/webpack.config.js:30]: out of curiosity,
module.exports에 ts, tsx 파일들 추가해주면 어떤 효과가 있어?

File: extension/src/html/index.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
```
Comment (sungho-cho) [inline extension/src/html/index.html:1]: popup 을 index로 옮긴 이유는 뭐야?
사실 이름만 바꾼거라 문제는 없을 것 같은데 그냥 궁금해서ㅎㅎ

File: extension/src/ts/background.ts [hunk 3]
```diff
@@ -2,27 +2,19 @@ import wretch from 'wretch';
...
-                });
+                textToSummarize = request.text;
             }
             else if (request.type === MsgType.PopUpInit) {
-                if (summarizedText != null) {
+                if (textToSummarize != null) {
                     sendResponse({
-                        summary: summarizedText
+                        textToSummarize: textToSummarize,
+                        articleTitle: "'Article Title Placeholder'"
```
Comment (sungho-cho) [inline extension/src/ts/background.ts:18]: nit: TODO plz :)

File: extension/src/ts/popup.tsx [hunk 4]
```diff
@@ -1,28 +1,74 @@
...
+            </AppBar>
+            <Card sx={{ height: 400 }}>
+                <CardContent>
+                    <Typography sx={{ fontSize: 18 }}>
+                        Summary of {articleTitle}: 
+                    </Typography>
+                </CardContent>
+                <CardContent>
+                    {(isLoading)
+                        ? <CircularProgress />
```
Comment (sungho-cho) [inline extension/src/ts/popup.tsx:65]: I'll look into this after this PR is merged. 
I might use React 18's new feature Suspense
https://www.daleseo.com/react-suspense/

**Review by sungho-cho — APPROVED** [review]
Approving with a few minor comments
Nice work as always 😎

File: extension/webpack.config.js (hunk 1 above)
Comment (junhur) [reply extension/webpack.config.js:30]: 나도 정확한 이해를 하고 있는건 아닌데 
module.exports안에 있는 entry object에 ts / tsx 파일 추가해서 module을 적용해서 output에 맞는 format으로 
/dist에다가 ts->js conversion을 해주는듯!

그래서 만약에 ts/tsx파일이 많아지면 하나하나 넣어주는거 보다 dynamically detect & include하려는 의도였어 ㅎㅎ

File: extension/src/html/index.html (hunk 2 above)
Comment (junhur) [reply extension/src/html/index.html:1]: 기존에 있던 popup.tsx에서 component + html injection logic이 같이 있어서 
좀 더 react convention 스럽게 component as its own file + typical naming convention (index as entry)하려고 refactor했어~

File: extension/src/ts/popup.tsx (hunk 4 above)
Comment (junhur) [reply extension/src/ts/popup.tsx:65]: 오... 이거 신박하다

File: extension/webpack.config.js (hunk 1 above)
Comment (sungho-cho) [reply extension/webpack.config.js:30]: 아하 js로 바꿔줘야 됐었지 맞네. 땡큐!

File: extension/src/html/index.html (hunk 2 above)
Comment (sungho-cho) [reply extension/src/html/index.html:1]: 오케이 이해해쓰~

**Pull request merged**